File: scripts/install_adapters.py
"""Install generated Humanizer adapter files into local tool locations."""

from __future__ import annotations

import argparse
import logging
import shutil
import subprocess
import sys
from pathlib import Path

LOGGER = logging.getLogger(__name__)
ROOT_DIR = Path(__file__).resolve().parent.parent


def install_file(source: Path, destination_dir: Path, destination_name: str) -> None:
    """Copy a source file into a destination directory."""
    if not source.exists():
        LOGGER.warning("Source not found: %s", source)
        return

    destination_dir.mkdir(parents=True, exist_ok=True)
    shutil.copy2(source, destination_dir / destination_name)


def parse_args() -> argparse.Namespace:
    """Parse CLI arguments."""
    parser = argparse.ArgumentParser(description=__doc__)
    parser.add_argument(
        "--skip-validation",
        action="store_true",
        help="Skip adapter validation before installation.",
    )
    return parser.parse_args()


def run_validation() -> None:
    """Validate adapters before installation."""
    result = subprocess.run(  # noqa: S603
        [sys.executable, "-m", "scripts.validate_adapters"],
        capture_output=True,
        text=True,
        check=False,
    )
    if result.returncode != 0:
        LOGGER.error("Validation failed: %s", result.stderr.strip() or "unknown error")
        raise SystemExit(1)


def main() -> None:
    """Install adapter files into local tool directories."""
    logging.basicConfig(level=logging.INFO)
    args = parse_args()

    if not args.skip_validation:
        run_validation()

    home = Path.home()
    source_gemini = ROOT_DIR / "adapters" / "gemini-extension"
    gemini_extensions = home / ".gemini" / "extensions" / "humanizer"

    if not source_gemini.exists():
        LOGGER.warning("Source not found: %s", source_gemini)
    else:
        if gemini_extensions.exists():
            shutil.rmtree(gemini_extensions)
        shutil.copytree(source_gemini, gemini_extensions)

    installs = [
        (
            ROOT_DIR / "adapters" / "antigravity-skill" / "SKILL.md",
            ROOT_DIR / ".agent" / "skills" / "humanizer",
            "SKILL.md",
        ),
        (
            ROOT_DIR / "adapters" / "antigravity-skill" / "SKILL_PROFESSIONAL.md",
            ROOT_DIR / ".agent" / "skills" / "humanizer",
            "SKILL_PROFESSIONAL.md",
        ),
        (
            ROOT_DIR / "adapters" / "qwen-cli" / "QWEN.md",
            home / ".qwen" / "extensions" / "humanizer",
            "QWEN.md",
        ),
        (
            ROOT_DIR / "adapters" / "codex" / "CODEX.md",
            home / ".codex" / "extensions" / "humanizer",
            "CODEX.md",
        ),
        (
            ROOT_DIR / "adapters" / "copilot" / "COPILOT.md",
            home / ".copilot" / "extensions" / "humanizer",
            "COPILOT.md",
        ),
        (ROOT_DIR / "adapters" / "vscode" / "HUMANIZER.md", ROOT_DIR / ".vscode", "HUMANIZER.md"),
        (
            ROOT_DIR / "adapters" / "opencode" / "SKILL.md",
            home / ".opencode" / "extensions" / "humanizer",
            "SKILL.md",
        ),
    ]

    for source, destination_dir, destination_name in installs:
        install_file(source, destination_dir, destination_name)


if __name__ == "__main__":
    main()

File: scripts/install-adapters.js
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';
import os from 'os';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const ROOT_DIR = path.resolve(__dirname, '..');
const BUNDLED_SKILL = path.join(ROOT_DIR, 'dist', 'humanizer-pro.bundled.md');
const USER_HOME = os.homedir();

/**
 * Copy bundled adapter content to a destination.
 * @param {string} name
 * @param {string} destDir
 * @param {string} [destFilename='GEMINI.md']
 */
function installTo(name, destDir, destFilename = 'GEMINI.md') {
  console.log(`Installing ${name}...`);
  try {
    if (!fs.existsSync(destDir)) {
      fs.mkdirSync(destDir, { recursive: true });
    }

    const destPath = path.join(destDir, destFilename);

    // Use bundled skill if available, otherwise check for source adapter
    if (fs.existsSync(BUNDLED_SKILL)) {
      fs.copyFileSync(BUNDLED_SKILL, destPath);
      console.log(`  [OK] Installed bundled skill to: ${destPath}`);
    } else {
      console.error(`  [FAIL] Bundled skill not found. Run 'npm run build' first.`);
    }
  } catch (e) {
    const message = e instanceof Error ? e.message : String(e);
    console.error(`  [FAIL] Could not install ${name}: ${message}`);
  }
}

console.log('--- Universal Humanizer Installer ---');

// 1. Gemini CLI
installTo('Gemini CLI', path.join(USER_HOME, '.gemini', 'extensions', 'humanizer'), 'GEMINI.md');

// 2. Qwen CLI
installTo('Qwen CLI', path.join(USER_HOME, '.qwen', 'extensions', 'humanizer'), 'QWEN.md');

// 3. Codex CLI
installTo('Codex CLI', path.join(USER_HOME, '.codex', 'extensions', 'humanizer'), 'CODEX.md');

// 4. Google Antigravity (Workspace)
installTo('Antigravity Skill', path.join(ROOT_DIR, '.agent', 'skills', 'humanizer'), 'SKILL.md');

// 5. VS Code (Workspace - typically requires global snippet install but copying to local for now as per previous pattern)
// Note: VS Code snippets are JSON, this markdown file is for context.
installTo('VS Code Context', path.join(ROOT_DIR, '.vscode'), 'HUMANIZER.md');

console.log('--- Installation Complete ---');
